pkg/cgidmap/cri.go

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package cgidmap
+
+import (
+	"container/list"
+	"context"
+	"path/filepath"
+	"sync"
+
+	"github.com/cilium/tetragon/pkg/cgroups"
+	"github.com/cilium/tetragon/pkg/cri"
+	"github.com/cilium/tetragon/pkg/logger"
+)
+
+// code for resolving missing cgroup ids by quering the CRI
+
+const (
+	// if, for whatever reason, we cannot talk to the CRI we dont want the queue to grow
+	// unbounded. Hence, we only keep the last 128 unresolved ids added. If we manage to catch
+	// up, subsequent sync updates from the pod hooks will ensure that the ids that were dropped
+	// will be added if they are still alive.
+	maxUnmappedIds = 128
+)
+
+type criResolver struct {
+	mu            sync.Mutex
+	cond          sync.Cond
+	unresolvedIDs *list.List
+}
+
+type unmappedID struct {
+	podID  PodID
+	contID ContainerID
+}
+
+func (c *criResolver) enqeue(unmappedIDs []unmappedID) {
```

Comment:
this is a bit tricky to read, I would have appreciated maybe some comments like: in this case inputs bigger than maxQueue so recreating truncated queue. Fitting by removing first/last elements. Sorry I wasn't super familiar with container/list.

I discovered the `cond.Signal`/`cond.Wait` thing, I imagine there was no way to simplify this by using a Go channel? I imagine the issue is that it's not easy to manage the Go channel (remove entry, empty it, etc).

Reply:
I don't know a way to do this with go channels. Especially since what the code does is keep the more recent entries.